common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/utils/checkins/FakeCheckinsGenerator.java

```diff
@@ -0,0 +1,50 @@
+package app.coronawarn.server.common.persistence.service.utils.checkins;
+
+import static app.coronawarn.server.common.persistence.service.utils.checkins.FakeCheckinIntervalSpecification.*;
+
+import app.coronawarn.server.common.protocols.internal.pt.CheckIn;
+import com.google.protobuf.ByteString;
+import java.security.SecureRandom;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.springframework.stereotype.Component;
+
+@Component
+public class FakeCheckinsGenerator {
+
+  private final Random random = new SecureRandom();
+
+  /**
+   * For each checkin in the original list, create X amount of fake checkins and collect to one
+   * single list which is returned. X is equal to the method's second parameter.
+   *
+   * @param pepper Optional element used to compute the hashes of the fake checkins.
+   */
+  public List<CheckIn> generateFakeCheckins(List<CheckIn> checkins, int numberOfFakesToCreate,
+      final byte[] pepper) {
+    return checkins.stream()
+        .flatMap(original -> IntStream.range(0, numberOfFakesToCreate)
+            .mapToObj(counter -> buildFakeCheckin(original, String.valueOf(counter),
+                (pepper == null || pepper.length == 0) ? randomHashPepper() : pepper)))
```

Comment:
Shouldn't this be an error, if the pepper is missing?

Reply:
Moving the pepper as an app configuration passed from Vault. Should take care of this scenario